### PR TITLE
feat: add seed to `Expr|Series.sample`

### LINKED ISSUE #973: [Enh]: Random seed for `sample`
### We would like to learn about your use case. For example, if this feature is needed  to adopt Narwhals in an open source project, could you please enter the link to it below?

I would like to use the `sample` method that allows for reproducibility. However, there is no random state parameter

### Please describe the purpose of the new feature or describe the problem to solve.

Reproducibility

### Suggest a solution if possible.

New parameter

### If you have tried alternatives, please describe them below.

_No response_

### Additional information that may help us understand your needs.

_No response_

### PR DESCRIPTION
**Pull request by FBruzzesi**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues 

- Closes #973 

## Checklist

- [x] Code follows style guide (ruff)
- [x] Tests added 
- [x] Documented the changes

### CODE REVIEW

File: narwhals/_arrow/expr.py [hunk 1]
```diff
@@ -249,12 +249,18 @@ def arg_true(self) -> Self:
     def sample(
         self: Self,
         n: int | None = None,
-        fraction: float | None = None,
         *,
+        fraction: float | None = None,
         with_replacement: bool = False,
+        seed: int | None = None,
```
Comment (wd60622) [inline narwhals/_arrow/expr.py:255]: Think np.random.Generator can be included here

Comment (FBruzzesi) [reply narwhals/_arrow/expr.py:255]: Cool thanks! I somehow keep forgetting about that! Just changed it

**Review by MarcoGorelli — APPROVED** [review]
nice one, thanks @FBruzzesi , and @wd60622 for the request!

looks good to me - @wd60622 if you could confirm that this would work for you, then we can ship it

**Review by wd60622 — APPROVED** [review]
You guys are fast! 🎉
Looks good

**Comment by MarcoGorelli** [comment]
thanks @wd60622 ! new version coming tomorrow morning 🚀 

if you're looking at using Narwhals in any Bayesian project please do feel free to ping us, we're both fans 😍

**Pull request merged**